packages/nodes-base/nodes/Zoho/GenericFunctions.ts

```diff
@@ -225,44 +226,71 @@ const adjustAccountIdField = adjustIdField('accountId', 'Account_Name');
 const adjustContactIdField = adjustIdField('contactId', 'Full_Name');
 const adjustDealIdField = adjustIdField('dealId', 'Deal_Name');
 
+const adjustCustomFields = (allFields: AllFields) => {
```

Comment:
This function isn't called for `product` resource, that's why the custom field doesn't get added to the record.